src/qt/transactiondesc.cpp

```diff
@@ -109,8 +109,24 @@ QString TransactionDesc::toHTML(CWallet *wallet, CWalletTx &wtx)
         strHTML += "<b>" + tr("Source") + ":</b> " + tr("Generated in CoinBase") + "<br>";
 
     else if (wtx.IsCoinStake())
-        strHTML += "<b>" + tr("Source") + ":</b> " + tr("Generated, PoS") + "<br>";
+    {
+        // Update support for Side Stake and correctly show POS/POR as well
+        strHTML += "<b>" + tr("Source") + ":</b> ";
+
+        MinedType gentype = GenerateType(wtx.GetHash(), vout);
 
+        switch (gentype)
+        {
+            case MinedType::POS               :    strHTML += tr("MINED - POS");        break;
+            case MinedType::POR               :    strHTML += tr("MINED - POR");        break;
+            case MinedType::ORPHANED          :    strHTML += tr("MINED - ORPHANED");   break;
+            case MinedType::POS_SIDE_STAKE    :    strHTML += tr("POS SIDE STAKE");     break;
+            case MinedType::POR_SIDE_STAKE    :    strHTML += tr("POR SIDE STAKE");     break;
```

Comment:
When MINED - POS, this should be SIDE STAKE - POS

Reply:
No, because the side stake is only on the output that has an address that is different than the stake input. The stake return to the original miner on the original address of the staking UTXO input is not a side stake. We have discussed creating a different category for returns to the original staker which is the remainder, after the distribution of the side stake to the receiving address. This would require creating two more enum categories to handle that.

Reply:
Is it right when I say, The POS_SIDE_STAKE (and por) is shown only in wallet of the receiver of the sidestake? I do not understand why it should be worded in different word order than MINED - POS (and por).

Reply:
Yes. POS_SIDE_STAKE and POR_SIDE_STAKE is only shown on the receiving wallet. The remainder which returns to the source wallet after the side stake distribution is simply labeled POS or POR... I will post pictures...

Reply:
The reasoning is that the remainder that returns is just like a normal stake return, except it is a lower amount because sidestakes have been taken out. This return is not to another address, hence the return itself is not a sidestake.

A strong argument can be made to create two more entries in the enum to handle the stake returns after a POS or POR sidestake has been taken out, to indicate that the full amount is not coming back to the staking address. I am open for that, and we would use the gold or purple cube with the red top to indicate that, much like we are using the gold cube with the green top to indicate the receiver of the side stake.

Reply:
It isn't necessary though, and I rather would do this in an additional PR, probably coupled with the UI work rather than mess further with it here... We need to get Camilla out of the door.

Reply:
![image](https://user-images.githubusercontent.com/7529186/52904250-5ab4eb00-31f7-11e9-8356-2dd8ca5d88a5.png)
![image](https://user-images.githubusercontent.com/7529186/52904254-715b4200-31f7-11e9-9e78-22076ca68146.png)

Reply:
![sidestake source](https://user-images.githubusercontent.com/7529186/52904288-db73e700-31f7-11e9-8754-37c6dead550a.jpg)
![sidestake destination](https://user-images.githubusercontent.com/7529186/52904289-ddd64100-31f7-11e9-801e-be6ee691496d.jpg)

Reply:
The red boxes are the stake returns to the source wallet for a POR and POS style stake. (The reason that wallet had both a POR and POS is the minimum block span between two consecutive research rewards was not satisfied.

The green boxes are the destination wallet of the sidestaking. In this particular case, there are two different sidestake entries pointing to different addresses in the same wallet. It shows separate transactions as intended...